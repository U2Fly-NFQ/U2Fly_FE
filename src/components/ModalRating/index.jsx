import React, { useState } from 'react'
import sadIcon from '../../assets/icons/boring.gif'
import happyIcon from '../../assets/icons/happy.gif'
import normalIcon from '../../assets/icons/normal.gif'
import loveIcon from '../../assets/icons/love.gif'
import cryingIcon from '../../assets/icons/crying.gif'
import { Modal, Tooltip, Input } from 'antd'
import './index.scss'
import { message } from 'antd/es'

const { TextArea } = Input

export default function ModalRating({ visible, setIsModalVisible, rating }) {
  const [status, setStatus] = useState()
  const [description, setDescription] = useState()
  const handleOK = () => {
    if (!status) {
      message.error('Please choose your emotion')
      return false
    }
    rating({
      status,
      description,
    })
    handleCancel()
  }
  const handleCancel = () => {
    setStatus('')
    setDescription('')
    setIsModalVisible(false)
  }
  return (
    <Modal visible={visible} onOk={handleOK} onCancel={handleCancel}>
      <div className="modal-rating">
        <div className="modal-rating__title">
          <h3>How do you feel about this trip ?</h3>
        </div>
        <div className="modal-rating__items">
          <div className="modal-rating__image">
            <Tooltip title="too bad">
              <img
                src={cryingIcon}
                alt=""
                onClick={() => {
                  setStatus(0)
                }}
                style={{
                  filter: status !== 0 ? 'grayscale(100%)' : 'grayscale(0)',
                }}
              />
            </Tooltip>
          </div>
          <div className="modal-rating__image">
            <Tooltip title="sad">
              <img
                src={sadIcon}
                alt=""
                onClick={() => {
                  setStatus(1)
                }}
                style={{
                  filter: status !== 1 ? 'grayscale(100%)' : 'grayscale(0)',
                }}
              />
            </Tooltip>
          </div>
          <div className="modal-rating__image">
            <Tooltip title="normal">
              <img
                src={normalIcon}
                alt=""
                onClick={() => {
                  setStatus(2)
                }}
                style={{
                  filter: status !== 2 ? 'grayscale(100%)' : 'grayscale(0)',
                }}
              />
            </Tooltip>
          </div>
          <div className="modal-rating__image">
            <Tooltip title="happy">
              <img
                src={happyIcon}
                alt=""
                onClick={() => {
                  setStatus(3)
                }}
                style={{
                  filter: status !== 3 ? 'grayscale(100%)' : 'grayscale(0)',
                }}
              />
            </Tooltip>
          </div>
          <div className="modal-rating__image">
            <Tooltip title="love">
              <img
                src={loveIcon}
                alt=""
                onClick={() => {
                  setStatus(4)
                }}
                style={{
                  filter: status !== 4 ? 'grayscale(100%)' : 'grayscale(0)',
                }}
              />
            </Tooltip>
          </div>
        </div>
      </div>
      <div className="modal-rating__description">
        <div className="modal-rating__title">
          <h3>Description</h3>
        </div>
        <div>
          <TextArea
            rows={4}
            placeholder="Enter your comment here!"
            value={description}
            onChange={(e) => {
              setDescription(e.target.value)
            }}
          />
        </div>
      </div>
    </Modal>
  )
}
