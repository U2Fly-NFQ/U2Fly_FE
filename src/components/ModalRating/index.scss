.modal-rating__items {
  display: flex;
  gap: 0.6rem;
  justify-content: center;
  .modal-rating__image {
    img {
      width: 50px;
      filter: grayscale(100%);
      cursor: pointer;
      transition: all 0.3s ease;
    }
    img:hover {
      filter: grayscale(0);
      transform: scale(1.2);
    }
    width: 50px;
  }
}
