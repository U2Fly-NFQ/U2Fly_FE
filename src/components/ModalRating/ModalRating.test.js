import { fireEvent, render, screen } from '@testing-library/react'
import React, { Suspense } from 'react'
import { BrowserRouter as Router } from 'react-router-dom'
import renderer from 'react-test-renderer'
import ModalRating from './'
describe('Modal rating', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Router>
          <Suspense>
            <ModalRating visible={true} />
          </Suspense>
        </Router>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('close when cancel', async () => {
    render(
      <Router>
        <Suspense>
          <ModalRating visible={true} setIsModalVisible={() => {}} rating={5} />
        </Suspense>
      </Router>
    )

    const ratingModal = await screen.findByTestId('rating-modal')
    const commentBox = await screen.findByTestId('comment-box')

    fireEvent.change(commentBox, {
      target: {
        value: 'comment',
      },
    })
    expect(commentBox.value).toBe('comment')

    fireEvent.click(screen.queryByText('Cancel'))
    expect(ratingModal).toBeInTheDocument()
  })
})
