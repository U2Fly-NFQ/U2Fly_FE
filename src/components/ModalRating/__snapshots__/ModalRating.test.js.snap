// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal rating  renders correctly 1`] = `null`;
